python/az/aro/azext_aro/_dynamic_validators.py

```diff
@@ -0,0 +1,335 @@
+        master_subnet = namespace.master_subnet
+        worker_subnet = namespace.worker_subnet
+        pod_cidr = namespace.pod_cidr
+        service_cidr = namespace.service_cidr
+
+        worker_parts = parse_resource_id(worker_subnet)
+        master_parts = parse_resource_id(master_subnet)
+
+        fn = validate_cidr("pod_cidr")
+        fn(namespace)
+        fn = validate_cidr("service_cidr")
+        fn(namespace)
+
+        cidr_array = {}
+
+        if pod_cidr is not None:
+            node_mask = 23 - int(pod_cidr.split("/")[1])
+            if node_mask < 2:
+                addresses.append(["Pod CIDR",
+                                  "Pod CIDR Capacity",
+                                  f"{pod_cidr} does not contain enough addresses for 3 master nodes " +
+                                  "(Requires cidr prefix of 21 or lower)"])
+            cidr_array["Pod CIDR"] = ipaddress.IPv4Network(pod_cidr)
+        if service_cidr is not None:
+            cidr_array["Service CIDR"] = ipaddress.IPv4Network(service_cidr)
+
+        network_client = get_mgmt_service_client(
+            cmd.cli_ctx, ResourceType.MGMT_NETWORK)
+
+        worker_subnet_obj = get_subnet(network_client, worker_subnet, worker_parts)
+
+        if worker_subnet_obj.address_prefix is None:
+            for address in worker_subnet_obj.address_prefixes:
+                cidr_array["Worker Subnet CIDR -- " + address] = ipaddress.IPv4Network(address)
+        else:
+            cidr_array["Worker Subnet CIDR"] = ipaddress.IPv4Network(worker_subnet_obj.address_prefix)
+
+        master_subnet_obj = get_subnet(network_client, master_subnet, master_parts)
+
+        if master_subnet_obj.address_prefix is None:
+            for address in master_subnet_obj.address_prefixes:
+                cidr_array["Master Subnet CIDR -- " + address] = ipaddress.IPv4Network(address)
+        else:
+            cidr_array["Master Subnet CIDR"] = ipaddress.IPv4Network(master_subnet_obj.address_prefix)
+
+        ipv4_zero = ipaddress.IPv4Network("0.0.0.0/0")
+
+        for item in cidr_array.items():
+            key = item[0]
+            cidr = item[1]
+            if not cidr.overlaps(ipv4_zero):
+                addresses.append([ERROR_KEY, key, f"{cidr} is not valid as it does not overlap with {ipv4_zero}"])
+            for item2 in cidr_array.items():
+                compare = item2[1]
+                if cidr is not compare:
+                    if cidr.overlaps(compare):
+                        addresses.append([ERROR_KEY, key, f"{cidr} is not valid as it overlaps with {compare}"])
+
+        hook.end()
+
+        return addresses
+
+    return _validate_cidr_ranges
+
+
+def dyn_validate_resource_permissions(service_principle_ids, resources):
+    def _validate_resource_permissions(cmd,
+                                       namespace):  # pylint: disable=unused-argument
+        errors = []
+
+        hook = cmd.cli_ctx.get_progress_controller()
```

Comment:
I see this hook functionality being repeated a bit. It's looks like it would be a great candidate for a decorator function like 
```
@shows_progress('Validating resource permissions')
def _validate_resource_permissions(cmd, namespace):
```